tests/test_compilers_default.py

```diff
@@ -162,7 +161,22 @@ def test_align_delay_measurement(platform, delay):
 
     target_sequence = PulseSequence()
     if delay > 0:
-        target_sequence[platform.qubits[0].probe.name].append(Delay(duration=delay))
-    target_sequence.extend(platform.qubits[0].native_gates.MZ.create_sequence())
+        target_sequence.append((platform.qubits[0].probe.name, Delay(duration=delay)))
+    target_sequence.concatenate(platform.qubits[0].native_gates.MZ.create_sequence())
     assert sequence == target_sequence
     assert len(sequence.probe_pulses) == 1
+
+
+def test_align_multiqubit(platform):
+    main, coupled = 0, 2
+    circuit = Circuit(3)
+    circuit.add(gates.GPI2(main, phi=0.2))
+    circuit.add(gates.CZ(main, coupled))
+    circuit.add(gates.M(main, coupled))
```

Comment:
This still fails if I change it to
```suggestion
    circuit.add(gates.M(main))
    circuit.add(gates.M(coupled))
```
which is generally supported by qibo, even though it should be equivalent. Particularly the delay on `main` is still 40ns instead of 70ns.

Reply:
This is a different story, and it seems to me that the problem is also present in 0.1.

Indeed, here the problem arises because qubit 0 has no channel involved beyond 40ns from the start, thus it is (in principle) free for further operations.
However, it is not truly free, because it is involved in a 2-qubits gate with qubit 2, even though no channel of qubit 0 is receiving a pulse.

To me, the most convenient way to solve the issue is to keep the compiler as it is, but add a delay in the native (we could even add automatically in the class, while creating the sequence - such that all qubits involved in a gate are padded, e.g. on the drive channel).

Reply:
There is even a further problem, fully specific to measurement: you may want to measure everything at the very end, in order to avoid non-unitary intermediate gates (possibly destroying the state of the other qubits, through entanglement).

But in this case, I expect an `Align` by the end user. And the best thing we can do is just to document it.

Reply:
> To me, the most convenient way to solve the issue is to keep the compiler as it is, but add a delay in the native (we could even add automatically in the class, while creating the sequence - such that all qubits involved in a gate are padded, e.g. on the drive channel).

I think that is an acceptable solution. The only potential issue could be if someone wants to actually do something in parallel to the two-qubit gate, for example for some weird calibration, but that is unlikely and if needed there are ways around it (define custom sequence without the delays). Following the comment below, feel free to open an issue and merge this PR, since it is certainly not related. Just noting that if we add the delay in the JSON (not automatically in the class), we will need a way to serialize it. I think we have not done this yet, but I assume it will be simple (a pulse with only `duration` would probably be interpreted as delay).

> There is even a further problem, fully specific to measurement: you may want to measure everything at the very end, in order to avoid non-unitary intermediate gates (possibly destroying the state of the other qubits, through entanglement).
> 
> But in this case, I expect an `Align` by the end user. And the best thing we can do is just to document it. (since in general mid-circuit measurements should be supported)

I have not tested what the compiler does for mid-circuit measurements, but I would also say that they should be supported since they are in qibo simulation. If the user does it, he is probably aware of the possibility to destroy the state, etc..

Reply:
> I think that is an acceptable solution. The only potential issue could be if someone wants to actually do something in parallel to the two-qubit gate, for example for some weird calibration, but that is unlikely and if needed there are ways around it (define custom sequence without the delays).

Yes, let's worry when we'll get an actual instance.

> Following the comment below, feel free to open an issue and merge this PR, since it is certainly not related.

Done, it's #976.

> Just noting that if we add the delay in the JSON (not automatically in the class), we will need a way to serialize it. I think we have not done this yet, but I assume it will be simple (a pulse with only `duration` would probably be interpreted as delay).

It's already serializable, by Pydantic (the `PulseLike` union is also needed for this purpose).